src/main/java/org/alliancegenome/curation_api/dao/GenomicEntityDAO.java

```diff
@@ -1,15 +1,43 @@
 package org.alliancegenome.curation_api.dao;
 
+import jakarta.persistence.Query;
 import org.alliancegenome.curation_api.dao.base.BaseSQLDAO;
 import org.alliancegenome.curation_api.model.entities.GenomicEntity;
 
 import jakarta.enterprise.context.ApplicationScoped;
 
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
 @ApplicationScoped
 public class GenomicEntityDAO extends BaseSQLDAO<GenomicEntity> {
 
 	protected GenomicEntityDAO() {
 		super(GenomicEntity.class);
 	}
 
+	public Map<String, Long> getGenomicEntityIdMap() {
+		if (genomicEntityIdMap.size() > 0) {
+			return genomicEntityIdMap;
+		}
+		Query q = entityManager.createNativeQuery("SELECT a.id, a.modEntityId, a.modInternalId FROM BiologicalEntity as a where exists (select * from genomicentity as g where g.id = a.id)");
+		List<Object[]> ids = q.getResultList();
+		ids.forEach(record -> {
+			if (record[1] != null) {
+				genomicEntityIdMap.put((String) record[1], (long)record[0]);
+			}
+			if (record[2] != null) {
+				genomicEntityIdMap.put((String) record[2], (long)record[0]);
```

Comment:
🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'typecast' is not followed by whitespace.
<!-- __reviewdog__:ChBhMjE4N2E3ZDE3NDFkZDJhEgpjaGVja3N0eWxl -->